### PR TITLE
Add support for the Jazz programming language.

### PR DESCRIPTION
**Pull request by frayment**
Updated grammars.yml with language support.
Updated languages.yml with language support.
Added language samples to `samples/Jazz`.
Added current version of repository `language-jazz` as a grammar.

"Out-in-the-wild" example (Still being worked on): [`jazz`](https://github.com/frayment/jazz)

### CODE REVIEW

**Comment by frayment** [comment]
Well judging by the fact that my language passed just fine, the build fail has nothing to do with me. So this Pull Request is staying open.

**Comment by Alhadis** [comment]
@frayment Alphabetical order needs to be taken into consideration. Please read [the contributor notes](https://github.com/github/linguist/blob/master/CONTRIBUTING.md) before resubmitting.

**Comment by frayment** [comment]
Hi @Alhadis. Long day, and just realised that it **is** my fault for the build fail. Fixing it up now. I don't understand what you mean about alphabetical order though?

**Comment by Alhadis** [comment]
[In the build feedback, here](https://travis-ci.org/github/linguist/jobs/133655772#L1150):
> TestPedantic#test_language_names_are_sorted [/home/travis/build/github/linguist/test/test_pedantic.rb:9]:
> Julia should come after Jazz

[You've listed the two out of order](https://github.com/frayment/linguist/blob/53529938f4e5139c1a41f72e3e316653274869ca/lib/linguist/languages.yml#L1782-L1795).

[Here's a link to in-the-wild usage for Jazz](https://github.com/search?q=extension%3Ajazz+NOT+nothack&type=Code).

**Comment by frayment** [comment]
@Alhadis Oh yes. I'm fixing that now. Thankyou. If you also look at the build information, it says:

> - MAXScript (#00a6a6) is too close to ["36b3b5", "00a6a6"]

Is this going to affect my build??

**Comment by Alhadis** [comment]
Also, you don't need to submit another pull request to make additional changes. Any additional commits you push to your topic branch will be included (assuming the branch hasn't already been merged).

> Is this going to affect my build??
I was going to ask about that. @arfon Why is this triggering a build failure? [I already modified](https://github.com/github/linguist/pull/2749/commits/f46e0536330e5b5bfd008169e2655d4f04654306) MAXScript's colour to avoid this, and this error is resurfacing again? FYI, the colour of #00AAAA is the colour used by Autodesk's own branding, so it wouldn't make sense to use anything else.

**Comment by frayment** [comment]
Thankyou. Is it possible to **delete** old ones? I've already opened two... 0_0

**Comment by Alhadis** [comment]
No, only if you're a maintainer. They'll just have to sit there until somebody deletes them.

**Comment by frayment** [comment]
Alright, thankyou for the information. First time doing this. ;)

**Comment by Alhadis** [comment]
Ah, I just noticed the colour error is being triggered because if the colour *you're* adding.

Yeah, you'll need to adjust the colour so it isn't too similar to existing colours. Currently, Jazz's colour is too similar to MAXScript and Dart.

**Comment by frayment** [comment]
Alright. Is there a way to calculate the colours available instead of having to go through everything? I could use the test file but grabbing the actual ranges available?

**Comment by Alhadis** [comment]
I don't think so. The way I did it was I opened up Photoshop and manually cranked the brightness level of the colour until it passed. This is probably an enhancement that could be included by the colour-tests...

**Comment by frayment** [comment]
Alright, thanks for the support, nice to have people like you available to help.

**Comment by frayment** [comment]
@Alhadis Uhhh also, I was just reading back over what you said, that "In the wild usage" thing, well I'm developing the language myself... From what I see, those usages are just people making up their own language (like me!). I looked at two examples of it and they are completely different... Infact one group of people who programmed some "jazz" doesn't even have members anymore.... Do you think this is going to affect me?

**Comment by frayment** [comment]
Reopening request as fixes have been modified.

**Comment by Alhadis** [comment]
Alright, there's no gentle way to put this: if there's not enough people using your language, it won't be accepted into GitHub Linguist.

I'm not a staff member, so I can't give you any official reasoning why. If I were to guess, however, I'd say the program's Bayesian classifier needs a sizeable number of results to work off before it can begin analysing languages based on their makeup and syntax.

So in other words, if there are less than a thousand *distributed* results on GitHub, it won't be enough. Emphasis on the word distributed: I don't think the language's own developer qualifies.

**Comment by frayment** [comment]
Hahaha, alright, alright. I understand your point. I guess I'll just leave the pull-request open, maybe change the name later on if it needs, until there are actually enough results.
